test/session_tests/DocumentQueryTest.ts

```diff
@@ -8,220 +8,493 @@ import {RequestExecutor} from "../../src/Http/Request/RequestExecutor";
+          index += 1;
+          expect(obj.order).to.be.equal(index);
+        });
 
-      expect(results).to.have.lengthOf(1);
     });
 
-    it('should query by whereIn', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product
-      }).whereIn<string>('name', ['test101', 'test107', 'test106']).get();
-      
-      expect(results).to.have.lengthOf(4);      
+    it('should query with orderBy as string', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .orderBy<string>('order', 'ASC').get();
+
+      expect(results[1].order).to.be.equal(10);
+
     });
 
-    it('should query by startsWith', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product
-      }).whereStartsWith('name', 'n').get();
-      
-      expect(results[0].name).to.equals('new_testing');
+    it('should query with orderByDESC as long', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .orderBy<string>('order as long', 'DESC').get();
+
+      expect(results).to.have.lengthOf(10);
+
+      results.reverse().map(function(obj, index) {
+        index += 1;
+        expect(obj.order).to.be.equal(index);
+      });
+
     });
 
-    it('should query by endsWith', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product
-      }).whereEndsWith('name', '7').get();
-      
-      expect(results[0].name).to.equals('test107');
+    it('should query with whereGreaterThanOrEqual', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .whereGreaterThanOrEqual<number>('order',6, 'name', 'withNesting').get();
+
+      expect(results).to.have.lengthOf(10);
+
+        results.map(function(obj) {
+
+          expect(obj).to.include({'name': 'withNesting'});
+
+        });
+
     });
 
-    it('should fail query with non-existing index', async () => expect(
-        store.openSession().query({
-          indexName: 's'
-        }).where({'Tag': 'Products'}).get()
-      ).to.be.rejected
+    it('should query with whereLessThanOrEqual', async () => {
+      const results: Universal[] = await  store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .whereLessThanOrEqual<number>('order',6, 'name', 'withNesting').get();
+
+      expect(results).to.have.lengthOf(10);
+
+      results.map(function(obj) {
+
+        expect(obj).to.include({'name': 'withNesting'});
```

Comment:
Wrong assertions and redundant parameters for `whereLessThanOrEqual` - see comment above.